streaming/stream_transfer_task.cc

```diff
@@ -239,8 +239,11 @@ future<> stream_transfer_task::execute() {
     });
 }
 
-void stream_transfer_task::append_ranges(const dht::token_range_vector& ranges) {
-    _ranges.insert(_ranges.end(), ranges.begin(), ranges.end());
```

Comment:
`std::vector::insert` already gives a strong exception guarantee, I think it's harder to see that the new code also gives it (better to rely on stdlib guarantees than enforce it on our own).

https://en.cppreference.com/w/cpp/container/vector/insert

Reply:
I think that's doable